guide_drafts/model-derives.md

```diff
@@ -0,0 +1,926 @@
+    pub last_name: String,
+    pub email: String,
+}
+
+pub struct NewUser<'a> {
+    pub first_name: &'a str,
+    pub last_name: &'a str,
+    pub email: &'a str,
+ }
+
+impl <'a, 'insert, DB> Insertable<users::table, DB> for &'insert NewUser<'a>
+where DB: Backend,
+      (ColumnInsertValue<users::first_name, AsExpr<&'insert &'a str, users::first_name>>,
+      ColumnInsertValue<users::last_name, AsExpr<&'insert &'a str, users::last_name>>,
+      ColumnInsertValue<users::email, AsExpr<&'insert &'a str, users::email>>): InsertValues<DB>
+{
+    type Values =
+        (ColumnInsertValue<users::first_name, AsExpr<&'insert &'a str, users::first_name>>,
+        ColumnInsertValue<users::last_name, AsExpr<&'insert &'a str, users::last_name>>,
+        ColumnInsertValue<users::email, AsExpr<&'insert &'a str, users::email>>);
+
+    #[allow(non_shorthand_field_patterns)]
+    fn values(self) -> Self::Values {
+        use diesel::expression::{AsExpression, Expression};
+        use diesel::insertable::ColumnInsertValue;
+
+        let NewUser {
+            first_name: ref first_name,
+            last_name: ref last_name,
+            email: ref email } = *self;
+
+
+        // Here is the returned tuple of three `ColumnInsertValue`'s
+        (
+            ColumnInsertValue::Expression(
+                  users::first_name,
+                  AsExpression::<<users::first_name as Expression>::SqlType>::as_expression(first_name)
+            ),
+
+            ColumnInsertValue::Expression(
+                users::last_name,
+                AsExpression::<<users::last_name as Expression>::SqlType>::as_expression(last_name)
+            ),
+
+            ColumnInsertValue::Expression(
+                users::last_name,
+                AsExpression::<<users::email as Expression>::SqlType>::as_expression(email)
+            ),
+        )
+    }
+}
+
+impl <'a: 'insert, 'insert, Op> IntoInsertStatement<users::table, Op> for &'insert NewUser<'a> {
+    type InsertStatement = InsertStatement<users::table, Self, Op>;
+
+    fn into_insert_statement(self, target: users::table, operator: Op) -> Self::InsertStatement {
+            InsertStatement::no_returning_clause(target, self, operator)
+    }
+}
+```
+
+One interesting piece to note are the values of the tuple for the associated type, `Values`.
+An `InsertValue` is a tuple that consists of each column to be inserted as an `ColumnInsertValue`.
+The returned value of `values()` has been formatted to illustrate this point.
+This implementation is considerably complex and you wouldn't want to be typing this out for each `Insertable` struct.
+
+## <a name="identifiable"></a>Identifiable
+
+The `Identifiable` trait is useful when you need to set up database [Associations](#associations) or are using certain Diesel database update features.
+When using the `Indentifiable` on a struct,
+it means that struct can be identified on a single table in the database. 
+
+Your struct *must* have an `id` field on it.
```

Comment:
Doesn't that conflict with

> If your table's primary key is named differently, you can annotate the table with the attribute

Reply:
whoops! That will be omitted. The next line is what I was trying to convey.